android_vs_ios.md

```diff
@@ -0,0 +1,175 @@
+[Espresso](https://developer.android.com/training/testing/espresso/) is the official framework for UI testing. It can interact and verify UI elements as a user would (clicking, checking text, etc). UI tests are run in a separate process, but they can still directly access the code of the app that’s being tested.\
+[Mockito](https://site.mockito.org/) is widely used for test mocks.
+
+iOS runs [unit tests](https://developer.apple.com/documentation/xctest) on device/simulator, so it’s fine to call iOS methods in unit tests.\
+Apple offers [UI test target](https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/testing_with_xcode/chapters/09-ui_testing.html) for running UI tests. Similar to Espresso, it runs in a separate process and can interact and verify UI elements as a user would (clicking, checking text, etc). However, it cannot access the app code and is meant for blackbox testing. Due to this, it might be more preferable to run UI tests on a unit test target instead of UI test target. The most popular UI testing frameworks ([KIF](https://github.com/kif-framework/KIF) and [EarlGrey](https://github.com/google/EarlGrey/)) are run on unit test target.\
+iOS developers generally write all mocks manually, partly due to lack of a good mocking framework. There is a library called [Cuckoo](https://github.com/Brightify/Cuckoo) which seems to be slightly more limited version of Mockito.
+
+# Other stuff
+## Equivalent terminologies/classes/frameworks
+|Android/Kotlin | iOS/Swift|
+|---|---|
+|`val` | `let`|
+|`fun` | `func`|
+|`when` | `switch`|
+|`null` | `nil`|
+|`dp` | point (resolution)|
+|Constructor | Init|
+|Interface | Protocol|
+|Lambda | Closure|
+|Kotlin sealed class | Enum|
+|Activity | ViewController|
+|layout xml | xib|
+|ConstraintLayout | Auto Layout|
+|LinearLayout | UIStackView (but more limited)|
+|TextView | UILabel|
+|EditText | UITextField / UITextView|
+|ListView (deprecated, use RecyclerView) | UITableView|
+|RecyclerView | UICollectionView|
+|RxJava | RxSwift|
+|RxBinding | RxCocoa|
+|LiveData | Driver/BehaviorRelay|
+|Retrofit | Alamofire (simple network calls can be done easily without a library)|
+|Gson/Jackson/Moshi | Codable|
+|Picasso/Glide | Kingfisher|
+|Timber | CocoaLumberjack|
+
+## Miscellaneous tidbits
+Since Android is open source, sometimes it's very helpful to look into the source code of framework classes when debugging issues.
+
+AppCode (as of version 2018.2) is not much better than XCode, it’s definitely nowhere near the quality of other JetBrains IDEs like IntelliJ or Android Studio. It’s noticeably slower, can't find code usage sometimes, has issues autocompleting certain extensions, and generally feels unreliable when refactoring.
```

Comment:
I would add that there is no storyboard (layout) editing functionality in AppCode as of version 3.3. Therefore, employing AppCode in the toolchain means using it in conjunction with XCode, making it more of a powerful code editor than a full IDE.